### ISSUE TITLE
HTTP Request node version 4.1  - .map() function producing invalid JSON Body

### ISSUE THREAD

**Issue by skyfox675** [n8n-io/n8n#10180]
### Bug Description

Hi Team!

When testing a new workflow, I've found that the HTTP Request node (v 4.1) is producing invalid JSON when using the following settings along with a .map() to iterate over an unknown number of items.

### To Reproduce

Using the following settings:
- Method: `Post`
- URL: `Any`
- Authentication: `None`
- Send Query Parameters: `true`
- Specify Query Parameters: `Using Fields Below`
  - Name: `token`
  - Value: `<redacted>`
- Send Headers: `false`
- Send body: `true`
- Body Content Type: `JSON`
- Specify Body: `Using JSON`
- JSON: 
``` Javascript
{
    "template_id": "<redacted>",
    "test": "yes",
    "custom_webhook_url": "<redacted>",
    "metadata": {{ JSON.stringify(JSON.stringify({"twentycrm_company_id":$('Get Company via CompanyID').item.json["data"]["company"]["id"]})) }},
    "placeholder_fields": [
        {
            "api_key": "Date",
            "value": "{{$today.toLocaleString({month: 'long', day: 'numeric', year: 'numeric'})}}"
        },
        {
            "api_key": "Company Name",
            "value": "{{ $('Get Company via CompanyID').item.json["data"]["company"]["companyLegalName"] }}"
        }
    ],
    "signers": [
{{ $('Get Company via CompanyID').item.json["data"]["company"]["companyOwnerS"].map(function (owner) {
const signer = {
"name": owner.name.firstName+ " " + owner.name.lastName,
"email": owner.email,
"mobile": owner.phone1,
"auto_sign": "no",
"signature_request_delivery_method": "email",
"signed_document_delivery_method": "email",
"required_identification_methods": [ "email", "sms" ]
}

return signer
})
}}
        ,{
            "name": "<redacted>",
            "email": "<redacted>",
            "mobile": "<redacted>",
            "auto_sign": "yes",
            "signature_request_delivery_method": "email",
            "signed_document_delivery_method": "email"
        }
    ]
}
```

### Expected behavior

Using my example above, this should produce a list of "signers" that contains 2 objects produced by the .map():
``` Javascript
{
    "template_id": "<redacted>",
    "test": "yes",
    "custom_webhook_url": "<redacted>",
    "metadata": "{\"twentycrm_company_id\":\"<redacted>\"}",
    "placeholder_fields": [
        {
            "api_key": "Date",
            "value": "July 24, 2024"
        },
        {
            "api_key": "Company Name",
            "value": "<redacted>"
        }
    ],
    "signers": [
{
    "name": "<redacted>",
    "email": "<redacted>",
    "mobile": "<redacted>",
    "auto_sign": "no",
    "signature_request_delivery_method": "email",
    "signed_document_delivery_method": "email",
    "required_identification_methods": ["email", "sms"]
},
{
    "name": "<redacted>",
    "email": "<redacted>",
    "mobile": "<redacted>",
    "auto_sign": "no",
    "signature_request_delivery_method": "email",
    "signed_document_delivery_method": "email",
    "required_identification_methods": ["email", "sms"]
}
,{
            "name": "<redacted>",
            "email": "<redacted>",
            "mobile": "<redacted>",
            "auto_sign": "yes",
            "signature_request_delivery_method": "email",
            "signed_document_delivery_method": "email"
        }
    ]
}
```

However this is the current result being produced. The closing `]` on `signers.*.required_identification_methods` is being stripped from the resultant objects causing invalid JSON syntax:
``` Javascript
{
    "template_id": "<redacted>",
    "test": "yes",
    "custom_webhook_url": "<redacted>",
    "metadata": "{\"twentycrm_company_id\":\"<redacted>\"}",
    "placeholder_fields": [
        {
            "api_key": "Date",
            "value": "July 24, 2024"
        },
        {
            "api_key": "Company Name",
            "value": "<redacted>"
        }
    ],
    "signers": [
{
    "name": "<redacted>",
    "email": "<redacted>",
    "mobile": "<redacted>",
    "auto_sign": "no",
    "signature_request_delivery_method": "email",
    "signed_document_delivery_method": "email",
    "required_identification_methods": ["email", "sms"
},
{
    "name": "<redacted>",
    "email": "<redacted>",
    "mobile": "<redacted>",
    "auto_sign": "no",
    "signature_request_delivery_method": "email",
    "signed_document_delivery_method": "email",
    "required_identification_methods": ["email", "sms"
}
 ,{
            "name": "<redacted>",
            "email": "<redacted>",
            "mobile": "<redacted>",
            "auto_sign": "yes",
            "signature_request_delivery_method": "email",
            "signed_document_delivery_method": "email"
        }
    ]
}
```

### Operating System

docker.n8n.io/n8nio/n8n:1.52.0

### n8n Version

1.52.2

### Node.js Version

Same as docker.n8n.io/n8nio/n8n:1.52.0

### Database

SQLite (default)

### Execution mode

queue

**Comment by skyfox675** [comment]
Hi team, I've confirmed this exists on n8n 1.52.2 as of this morning

**Comment by skyfox675** [comment]
Also confirmed this exists on n8n 1.53.0 as of this morning as well

**Comment by Joffcom** [comment]
Hey @skyfox675,

Thanks for the report, We have created an internal ticket to get this resolved in a future release.

**Comment by michael-radency** [comment]
Hello @skyfox675, thanks for reporting
while we are working on fix you could wrap your function in` JSON.stringify ` to this fix issue
```
{{ JSON.stringify($('Get Company via CompanyID').item.json["data"]["company"]["companyOwnerS"].map(function (owner) {
const signer = {
"name": owner.name.firstName+ " " + owner.name.lastName,
"email": owner.email,
"mobile": owner.phone1,
"auto_sign": "no",
"signature_request_delivery_method": "email",
"signed_document_delivery_method": "email",
"required_identification_methods": [ "email", "sms" ]
}

return signer
}))
}}

```

**Comment by skyfox675** [comment]
@michael-radency so sorry for missing this! 
That's actually the solution I found. I moved the function into a Set node and in another case leveraged JSON.stringify tonight for the same issue. Appreciate the workaround in the meantime!

**Comment by janober** [comment]
Fix got released with [`n8n@1.64.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.64.0)
